src/stac_asset/client.py

```diff
@@ -226,15 +235,35 @@ async def get_client(self, href: str) -> Client:
             raise ValueError(f"could not guess client class for href: {href}")
 
         async with self.lock:
-            if client_class in self.clients:
-                return self.clients[client_class]
+            if client_class.name in self.clients:
+                return self.clients[client_class.name]
             else:
                 client = await client_class.from_config(self.config)
-                self.clients[client_class] = client
+                self.clients[client_class.name] = client
                 return client
 
     async def close_all(self) -> None:
         """Close all clients."""
         async with self.lock:
             for client in self.clients.values():
                 await client.close()
+
+
+def _get_client_class_by_name(name: str) -> type[Client]:
+    from .earthdata_client import EarthdataClient
+    from .filesystem_client import FilesystemClient
+    from .http_client import HttpClient
+    from .planetary_computer_client import PlanetaryComputerClient
+    from .s3_client import S3Client
+
+    client_classes: list[type[Client]] = [
+        EarthdataClient,
+        FilesystemClient,
+        HttpClient,
+        PlanetaryComputerClient,
+        S3Client,
+    ]
+    for client_class in client_classes:
+        if client_class.name == name:
+            return client_class
+    raise ValueError(f"no client with name: {name}")
```

Comment:
Needs more oregano, or similar spice to this:

```suggestion
    client_classes: dict[str, type[Client]] = {
        c.name: c for c in Client.__subclasses__()
    }

    try:
        return client_class[name]
    except KeyError:
        raise ValueError(f"no client with name: {name}")
```